File: sources/net.sf.j2s.core/test/src/com/falstad/circuit/WireElm.js
(function(){var P$=Clazz.newPackage("com.falstad.circuit"),I$=[];
var C$=Clazz.newClass(P$, "WireElm", null, 'com.falstad.circuit.CircuitElm');

C$.$clinit$ = function() {Clazz.load(C$, 1);
}

Clazz.newMeth(C$, '$init$', function () {
}, 1);

Clazz.newMeth(C$, 'c$$I$I', function (xx, yy) {
C$.superclazz.c$$I$I.apply(this, [xx, yy]);
C$.$init$.apply(this);
}, 1);

Clazz.newMeth(C$, 'c$$I$I$I$I$I$java_util_StringTokenizer', function (xa, ya, xb, yb, f, st) {
C$.superclazz.c$$I$I$I$I$I.apply(this, [xa, ya, xb, yb, f]);
C$.$init$.apply(this);
}, 1);

Clazz.newMeth(C$, 'draw$java_awt_Graphics', function (g) {
this.setVoltageColor$java_awt_Graphics$D(g, this.volts[0]);
P$.CircuitElm.drawThickLine$java_awt_Graphics$java_awt_Point$java_awt_Point(g, this.point1, this.point2);
this.doDots$java_awt_Graphics(g);
this.setBbox$java_awt_Point$java_awt_Point$D(this.point1, this.point2, 3);
if (this.mustShowCurrent()) {
var s = P$.CircuitElm.getShortUnitText$D$S(Math.abs(this.getCurrent()), "A");
this.drawValues$java_awt_Graphics$S$D(g, s, 4);
} else if (this.mustShowVoltage()) {
var s = P$.CircuitElm.getShortUnitText$D$S(this.volts[0], "V");
this.drawValues$java_awt_Graphics$S$D(g, s, 4);
}this.drawPosts$java_awt_Graphics(g);
});

Clazz.newMeth(C$, 'stamp', function () {
(I$[0]||(I$[0]=Clazz.load('com.falstad.circuit.CircuitElm'))).sim.stampVoltageSource$I$I$I$D(this.nodes[0], this.nodes[1], this.voltSource, 0);
});

Clazz.newMeth(C$, 'mustShowCurrent', function () {
return (this.flags & 1) != 0;
});

Clazz.newMeth(C$, 'mustShowVoltage', function () {
return (this.flags & 2) != 0;
});

Clazz.newMeth(C$, 'getVoltageSourceCount', function () {
return 1;
});

Clazz.newMeth(C$, 'getInfo$SA', function (arr) {
arr[0] = "wire";
arr[1] = "I = " + P$.CircuitElm.getCurrentDText$D(this.getCurrent());
arr[2] = "V = " + P$.CircuitElm.getVoltageText$D(this.volts[0]);
});

Clazz.newMeth(C$, 'getDumpType', function () {
return 'w'.$c();
});

Clazz.newMeth(C$, 'getPower', function () {
return 0;
});

Clazz.newMeth(C$, 'getVoltageDiff', function () {
return this.volts[0];
});

Clazz.newMeth(C$, 'isWire', function () {
return true;
});

Clazz.newMeth(C$, 'getEditInfo$I', function (n) {
if (n == 0) {
var ei = Clazz.new_((I$[1]||(I$[1]=Clazz.load('com.falstad.circuit.EditInfo'))).c$$S$D$D$D,["", 0, -1, -1]);
ei.checkbox = Clazz.new_((I$[2]||(I$[2]=Clazz.load('a2s.Checkbox'))).c$$S$Z,["Show Current", this.mustShowCurrent()]);
return ei;
}if (n == 1) {
var ei = Clazz.new_((I$[1]||(I$[1]=Clazz.load('com.falstad.circuit.EditInfo'))).c$$S$D$D$D,["", 0, -1, -1]);
ei.checkbox = Clazz.new_((I$[2]||(I$[2]=Clazz.load('a2s.Checkbox'))).c$$S$Z,["Show Voltage", this.mustShowVoltage()]);
return ei;
}return null;
});

Clazz.newMeth(C$, 'setEditValue$I$com_falstad_circuit_EditInfo', function (n, ei) {
if (n == 0) {
if (ei.checkbox.getState()) this.flags = 1;
 else this.flags = this.flags&(-2);
}if (n == 1) {
if (ei.checkbox.getState()) this.flags = 2;
 else this.flags = this.flags&(-3);
}});

Clazz.newMeth(C$, 'getShortcut', function () {
return 'w'.$c();
});

Clazz.newMeth(C$);
})();
//Created 2017-12-17 19:28:23

File: sources/net.sf.j2s.core/test/src/com/falstad/circuit/WireElm.java
package com.falstad.circuit;

import java.awt.Graphics;
import java.util.StringTokenizer;

import a2s.Checkbox;

    class WireElm extends CircuitElm {
	public WireElm(int xx, int yy) { super(xx, yy); }
	public WireElm(int xa, int ya, int xb, int yb, int f,
		       StringTokenizer st) {
	    super(xa, ya, xb, yb, f);
	}
	static final int FLAG_SHOWCURRENT = 1;
	static final int FLAG_SHOWVOLTAGE = 2;
	void draw(Graphics g) {
	    setVoltageColor(g, volts[0]);
	    drawThickLine(g, point1, point2);
	    doDots(g);
	    setBbox(point1, point2, 3);
	    if (mustShowCurrent()) {
	        String s = getShortUnitText(Math.abs(getCurrent()), "A");
	        drawValues(g, s, 4);
	    } else if (mustShowVoltage()) {
	        String s = getShortUnitText(volts[0], "V");
	        drawValues(g, s, 4);
	    }
	    drawPosts(g);
	}
	void stamp() {
	    sim.stampVoltageSource(nodes[0], nodes[1], voltSource, 0);
	}
	boolean mustShowCurrent() {
	    return (flags & FLAG_SHOWCURRENT) != 0;
	}
	boolean mustShowVoltage() {
	    return (flags & FLAG_SHOWVOLTAGE) != 0;
	}
	int getVoltageSourceCount() { return 1; }
	void getInfo(String arr[]) {
	    arr[0] = "wire";
	    arr[1] = "I = " + getCurrentDText(getCurrent());
	    arr[2] = "V = " + getVoltageText(volts[0]);
	}
	int getDumpType() { return 'w'; }
	double getPower() { return 0; }
	double getVoltageDiff() { return volts[0]; }
	boolean isWire() { return true; }
	public EditInfo getEditInfo(int n) {
	    if (n == 0) {
		EditInfo ei = new EditInfo("", 0, -1, -1);
		ei.checkbox = new Checkbox("Show Current", mustShowCurrent());
		return ei;
	    }
	    if (n == 1) {
		EditInfo ei = new EditInfo("", 0, -1, -1);
		ei.checkbox = new Checkbox("Show Voltage", mustShowVoltage());
		return ei;
	    }
	    return null;
	}
	public void setEditValue(int n, EditInfo ei) {
	    if (n == 0) {
		if (ei.checkbox.getState())
		    flags = FLAG_SHOWCURRENT;
		else
		    flags &= ~FLAG_SHOWCURRENT;
	    }
	    if (n == 1) {
		if (ei.checkbox.getState())
		    flags = FLAG_SHOWVOLTAGE;
		else
		    flags &= ~FLAG_SHOWVOLTAGE;
	    }
	}
        int getShortcut() { return 'w'; }
    }
